### ISSUE TITLE
[suggestion] shifting requiredSkillTries calculation/cap

### ISSUE THREAD

**Issue by Zbizu** [otland/forgottenserver#2950]
current problem:
- getRequiredSkillTries is calculated based on base value * vocation multiplier
- variable type limits max skill level in weird places

possible solution:
- a change to required skill tries calculation to `(base value * vocation multiplier) / skill rate`

pros:
- higher skill can be reached

cons:
- changing skill rate can ruin skillTries to level relation if using this solution

alternatively data type can be changed to increase skill cap, what do you think?

**Comment by EPuncker** [comment]
sorry but I didn't quite understand what you are proposing... do we have skill limit now, and you are proposing a change in calculation to make them higher, so we can achieve higher skills? if yes, we need to see the value that the client display without debugging

**Comment by Zbizu** [comment]
current calculation (multiple skill tries in one hit):
- `skillTries to level up = base tries needed to level * vocation mod`
- `skillTries gain = 1 * rateSkill`

my suggestion (one skill try in one hit but less tries needed to level up):
- `skillTries to level up = base tries needed to level * vocation mod / rateSkill`
- `skillTries gain = 1`

we can do number % 256 for displaying skill level in tibia client and change data type for other clients (eg. uint)

alternative solutions for storing big amount of skillTries in database:
- skill level, current skill %, skillTries done inside that 1%
or
- counting how many times skilltries went over the cap (second database field)
or
- split the total number into skillTries_big and skillTries_small and concat in lua (first google result said that lua can store 2^52 integer)

**Comment by Wirox** [comment]
> changing skill rate can ruin skillTries to level relation if using this solution

I believe this is enough of a problem not to implement your idea.

Another issue is that we would now have to either round required skill tries, or store it as a floating point number, both of which result in losing accuracy of calculations.

If there really is a need to increase the skill cap, I don't think there is a better solution than simply using a bigger integer data type.

**Comment by Zbizu** [comment]
> Another issue is that we would now have to either round required skill tries, or store it as a floating point number, both of which result in losing accuracy of calculations.

I admit that my proposed solution creates more problems than it solves, but we have a lot of great minds in this community so I decided to give it a try.

I did some maths in excel and concluded that if we stored per percent tries instead of per level and set tries to 0 everytime player levels up (or use some solution that scientists use for astronomical calculations), we could gain enough levels to go over 255. However, it's probably not worth to calculate that much with each skilltry when only a few servers would do high rates because it would hurt performance of servers with normal rates using extended skill cap solutions.

Closing as I don't see any good and precise way of doing this and it isn't that important to have.
